### ISSUE TITLE
to_string’ is not a member of ‘std

### ISSUE THREAD

**Issue by hufo1** [01org/mkl-dnn#160]
Linux17.10,g++-5.4,

when "make"  the file:
error: ‘to_string’ is not a member of ‘std’
                 L((".subloop20x" + std::to_string(i)).c_str());

i know it need C++11，but my g++ version is 5.4 
How to solve it?

**Comment by vpirogov** [comment]
We need slightly more information to investigate this. Could you please provide information about your environment and relevant logs as indicated in issue template?

**Comment by vpirogov** [comment]
Closing as with information provided there's no way to reproduce the issue. Please feel free to reopen with additional information.

**Comment by emicol** [comment]
I've the same problem

mkdir -p build && cd build && cmake .. && make
-- The C compiler identification is GNU 5.4.1
-- The CXX compiler identification is GNU 5.4.1
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- CMAKE_BUILD_TYPE is unset, defaulting to Release
-- Detecting Intel(R) MKL: trying mklml_intel
-- Intel(R) MKL: include /home/emilie/mkl-dnn/external/mklml_lnx_2018.0.1.20171007/include
-- Intel(R) MKL: lib /home/emilie/mkl-dnn/external/mklml_lnx_2018.0.1.20171007/lib/libmklml_intel.so
-- Intel(R) MKL: OpenMP lib /home/emilie/mkl-dnn/external/mklml_lnx_2018.0.1.20171007/lib/libiomp5.so
-- Found OpenMP_C: -fopenmp (found version "4.0") 
-- Found OpenMP_CXX: -fopenmp (found version "4.0") 
-- Found Doxygen: /usr/bin/doxygen (found version "1.8.13") found components:  doxygen dot 
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - found
-- Found Threads: TRUE  
-- Configuring done
-- Generating done
-- Build files have been written to: /home/emilie/mkl-dnn/build
Scanning dependencies of target mkldnn
[  0%] Building CXX object src/CMakeFiles/mkldnn.dir/common/batch_normalization.cpp.o
[  1%] Building CXX object src/CMakeFiles/mkldnn.dir/common/convolution.cpp.o
[  1%] Building CXX object src/CMakeFiles/mkldnn.dir/common/convolution_relu.cpp.o
[  2%] Building CXX object src/CMakeFiles/mkldnn.dir/common/eltwise.cpp.o
[  2%] Building CXX object src/CMakeFiles/mkldnn.dir/common/engine.cpp.o
[  3%] Building CXX object src/CMakeFiles/mkldnn.dir/common/inner_product.cpp.o
[  4%] Building CXX object src/CMakeFiles/mkldnn.dir/common/lrn.cpp.o
[  4%] Building CXX object src/CMakeFiles/mkldnn.dir/common/memory.cpp.o
[  5%] Building CXX object src/CMakeFiles/mkldnn.dir/common/memory_desc_wrapper.cpp.o
[  5%] Building CXX object src/CMakeFiles/mkldnn.dir/common/pooling.cpp.o
[  6%] Building CXX object src/CMakeFiles/mkldnn.dir/common/primitive.cpp.o
[  7%] Building CXX object src/CMakeFiles/mkldnn.dir/common/primitive_attr.cpp.o
[  7%] Building CXX object src/CMakeFiles/mkldnn.dir/common/primitive_desc.cpp.o
[  8%] Building CXX object src/CMakeFiles/mkldnn.dir/common/primitive_iterator.cpp.o
[  8%] Building CXX object src/CMakeFiles/mkldnn.dir/common/query.cpp.o
[  9%] Building CXX object src/CMakeFiles/mkldnn.dir/common/reorder.cpp.o
[  9%] Building CXX object src/CMakeFiles/mkldnn.dir/common/scratchpad.cpp.o
[ 10%] Building CXX object src/CMakeFiles/mkldnn.dir/common/softmax.cpp.o
[ 11%] Building CXX object src/CMakeFiles/mkldnn.dir/common/stream.cpp.o
[ 11%] Building CXX object src/CMakeFiles/mkldnn.dir/cpu/cpu_barrier.cpp.o
[ 12%] Building CXX object src/CMakeFiles/mkldnn.dir/cpu/cpu_engine.cpp.o
[ 12%] Building CXX object src/CMakeFiles/mkldnn.dir/cpu/cpu_reducer.cpp.o
[ 13%] Building CXX object src/CMakeFiles/mkldnn.dir/cpu/cpu_reorder.cpp.o
[ 14%] Building CXX object src/CMakeFiles/mkldnn.dir/cpu/gemm_convolution.cpp.o
[ 14%] Building CXX object src/CMakeFiles/mkldnn.dir/cpu/gemm_convolution_utils.cpp.o
[ 15%] Building CXX object src/CMakeFiles/mkldnn.dir/cpu/gemm_inner_product.cpp.o
[ 15%] Building CXX object src/CMakeFiles/mkldnn.dir/cpu/jit_avx2_1x1_conv_kernel_f32.cpp.o
[ 16%] Building CXX object src/CMakeFiles/mkldnn.dir/cpu/jit_avx2_1x1_convolution.cpp.o
[ 16%] Building CXX object src/CMakeFiles/mkldnn.dir/cpu/jit_avx2_conv_kernel_f32.cpp.o
[ 17%] Building CXX object src/CMakeFiles/mkldnn.dir/cpu/jit_avx2_convolution.cpp.o
[ 18%] Building CXX object src/CMakeFiles/mkldnn.dir/cpu/jit_avx2_gemm_f32.cpp.o
[ 18%] Building CXX object src/CMakeFiles/mkldnn.dir/cpu/jit_avx512_common_1x1_conv_kernel.cpp.o
[ 19%] Building CXX object src/CMakeFiles/mkldnn.dir/cpu/jit_avx512_common_1x1_convolution.cpp.o
[ 19%] Building CXX object src/CMakeFiles/mkldnn.dir/cpu/jit_avx512_common_conv_kernel.cpp.o
[ 20%] Building CXX object src/CMakeFiles/mkldnn.dir/cpu/jit_avx512_common_conv_winograd_kernel_f32.cpp.o
[ 21%] Building CXX object src/CMakeFiles/mkldnn.dir/cpu/jit_avx512_common_convolution.cpp.o
[ 21%] Building CXX object src/CMakeFiles/mkldnn.dir/cpu/jit_avx512_common_convolution_winograd.cpp.o
[ 22%] Building CXX object src/CMakeFiles/mkldnn.dir/cpu/jit_avx512_common_gemm_f32.cpp.o
/home/emilie/mkl-dnn/src/cpu/jit_avx512_common_gemm_f32.cpp: In lambda function:
/home/emilie/mkl-dnn/src/cpu/jit_avx512_common_gemm_f32.cpp:1324:36: error: ‘to_string’ is not a member of ‘std’
                 L((".subloop20x" + std::to_string(i)).c_str());
                                    ^
/home/emilie/mkl-dnn/src/cpu/jit_avx512_common_gemm_f32.cpp:1327:42: error: ‘to_string’ is not a member of ‘std’
                     jne((".subloop20x" + std::to_string(i + 1)).c_str(),
                                          ^
/home/emilie/mkl-dnn/src/cpu/jit_avx512_common_gemm_f32.cpp:1351:40: error: ‘to_string’ is not a member of ‘std’
                     L((".subloop30x" + std::to_string(i)).c_str());
                                        ^
/home/emilie/mkl-dnn/src/cpu/jit_avx512_common_gemm_f32.cpp:1354:46: error: ‘to_string’ is not a member of ‘std’
                         jne((".subloop30x" + std::to_string(i + 1)).c_str(),
                                              ^
/home/emilie/mkl-dnn/src/cpu/jit_avx512_common_gemm_f32.cpp:1424:36: error: ‘to_string’ is not a member of ‘std’
                 L((".subloop20x" + std::to_string(i) + "mask").c_str());
                                    ^
/home/emilie/mkl-dnn/src/cpu/jit_avx512_common_gemm_f32.cpp:1427:42: error: ‘to_string’ is not a member of ‘std’
                     jne((".subloop20x" + std::to_string(i + 1) + "mask")
                                          ^
/home/emilie/mkl-dnn/src/cpu/jit_avx512_common_gemm_f32.cpp:1452:40: error: ‘to_string’ is not a member of ‘std’
                     L((".subloop30x" + std::to_string(i) + "mask").c_str());
                                        ^
/home/emilie/mkl-dnn/src/cpu/jit_avx512_common_gemm_f32.cpp:1455:46: error: ‘to_string’ is not a member of ‘std’
                         jne((".subloop30x" + std::to_string(i + 1) + "mask")
                                              ^
src/CMakeFiles/mkldnn.dir/build.make:950 : la recette pour la cible « src/CMakeFiles/mkldnn.dir/cpu/jit_avx512_common_gemm_f32.cpp.o » a échouée
make[2]: *** [src/CMakeFiles/mkldnn.dir/cpu/jit_avx512_common_gemm_f32.cpp.o] Erreur 1
CMakeFiles/Makefile2:117 : la recette pour la cible « src/CMakeFiles/mkldnn.dir/all » a échouée
make[1]: *** [src/CMakeFiles/mkldnn.dir/all] Erreur 2
Makefile:140 : la recette pour la cible « all » a échouée
make: *** [all] Erreur 2

**Comment by emfomenk** [comment]
I will replace to_string with something else, and that should solve the problem. Thanks for reporting.
